packages/docusaurus-plugin-content-docs/src/sidebars/utils.ts

```diff
@@ -211,12 +213,16 @@ export function createSidebarsUtils(sidebars: Sidebars): SidebarsUtils {
     if (!sidebarName) {
       return emptySidebarNavigation();
     }
-    const navigationItems = sidebarNameToNavigationItems[sidebarName];
+    let navigationItems = sidebarNameToNavigationItems[sidebarName];
     if (!navigationItems) {
       throw new Error(
         `Doc with ID ${docId} wants to display sidebar ${sidebarName} but a sidebar with this name doesn't exist`,
       );
     }
+    navigationItems = navigationItems.filter(
+      (item) => !(item.type === 'doc' && unlistedIds.includes(item.id)),
+    );
```

Comment:
Weird edge-case to consider but:
- on siblings, we don't want. to navigate to an unlisted doc (ok)
- on the unlisted doc, don't we want to allow navigating to siblings?

CF:

https://deploy-preview-8004--docusaurus-2.netlify.app/tests/docs/test-unlisted

Does it make sense to allow navigating to next/previous here?

<img width="1858" alt="CleanShot 2022-09-08 at 18 16 24@2x" src="https://user-images.githubusercontent.com/749374/189173369-6129eed4-bde3-4ac9-930d-f5872e4ec9ae.png">

Reply:
This behavior looks fine 👍  and is covered by snapshot tests

2 consecutive unlisted docs should rather not allow navigating from one to another: accessing unlistedDoc1 may reveal the existence of unlistedDoc2